src/server/stock_market/stock_market_controller.py

```diff
@@ -8,9 +11,17 @@ class StockMarketController:
 
     @staticmethod
     @blueprint.route("/chart")
-    def chart() -> str:
+    def chart() -> Response:
         """Chart data retrieval endpoint."""
-        return "chart"
+        args = request.args
+        from_param: str | None = args.get("from")
```

Comment:
Please replace `str | None` with `Optional[str]`. I think this even can be simplified to:
```python
from_param: str = args.get("from", "")
...
if not from_param:
    return ...
```
`args.get` will also raise exception when `"from"` key does not exist in `args` dict, we can avoid that by specifying second argument.

Reply:
Done